### ISSUE TITLE
about koan "New processes are spawned functions"

### ISSUE THREAD

**Issue by Airmacho** [elixirkoans/elixir-koans#166]
I am a beginner of elixir learning. Thank you for this.

As to koan "New processes are spawned functions" in [15_processes.ex](https://github.com/elixirkoans/elixir-koans/blob/master/lib/koans/15_processes.ex#L20) file.

My iex(elixir 1.4.1) returns `false` when I use Process.alive?(pid)

**Comment by felipesere** [comment]
Thank you for reporting this issue!
I just updated to 1.4.1 and tried it myself and got through it normally.
Can you try 'mix test' which will run all the Koans with the appropriate answers?
If that fails too, we'll have to dig deeper...

**Comment by iamvery** [comment]
It seems the test expects the answer to be true (see bef1cf), but I also observed that it's false in the console. I'm pretty sure this is a timing problem. Since nothing in the spawn process is blocking, it's only alive for a _very short_ amount of time. The tests run really fast, so it gets to the alive check before the spawned process dies.

```elixir
iex(1)> pid = spawn(fn -> nil end); Process.alive?(pid)
true
iex(2)> pid = spawn(fn -> nil end)
#PID<0.88.0>
iex(3)> Process.alive?(pid)
false
```

So in general, this is a bad/confusing example, and we should consider updating it.

**Comment by felipesere** [comment]
Yeah, I was thinking the same and started experimenting with that on my laptop.
We could put a recursive function in there that will loop forever?

**Comment by iamvery** [comment]
Perhaps a `receive` call would be more idiomatic?

**Comment by felipesere** [comment]
Looks good 😄

**Comment by jessejanderson** [comment]
This seems backward to me... shouldn't `Process.alive?` return false to indicate that when a process has nothing to do, it dies immediately?

The `receive` example seems like a hack to me in the current context that teaches false information.

Maybe this should be split into 2 tests. The first is as written (maybe with a tweaked description) expecting an answer of `false`, then the next test has a receive and can return true?

**Comment by jessejanderson** [comment]
(maybe I'm just annoyed that I knew the process would be dead immediately, and the test was telling me that I was wrong) :D

**Comment by iamvery** [comment]
@jessejanderson that's a good point. To me it basically feels like "concurrency is hard. look concurrency is hard to teach too" 😆 

I'm not sure there is a reliable way to have the process die before the Process.alive?/1 check. I suppose that's a function of using procedural tests on concurrent code. Maybe the whole approach to this lesson needs to be freshened up? Seems with koans it's hard to design a coherent story by tweaking individual lessons. 🤔

**Comment by cybrox** [comment]
I think this can be closed now that #173 and #180 are merged. 😃    
cc @iamvery
